engine/src/test/java/org/hibernate/validator/test/internal/engine/path/NodeImplTest.java

```diff
@@ -181,6 +189,95 @@ public void testToMapCascadeValidate() {
 		assertEquals( node.getKind(), ElementKind.PROPERTY, "unexpected node kind" );
 	}
 
+	@Test
+	public void testPropertyNodeGetValueForSet() {
+		Building building = new Building();
+		building.apartments.add( new Apartment( new Person( "Bob" ) ) );
+		building.apartments.add( new Apartment( new Person( "Alice" ) ) );
+
+		Set<ConstraintViolation<Building>> constraintViolations = validator.validate( building );
+
+		assertNumberOfViolations( constraintViolations, 1 );
+		assertCorrectPropertyPaths( constraintViolations, "apartments[].resident.name" );
+
+		Path path = constraintViolations.iterator().next().getPropertyPath();
+		Iterator<Path.Node> nodeIterator = path.iterator();
+
+		Path.Node node = nodeIterator.next();
+		assertEquals( node.getKind(), ElementKind.PROPERTY, "unexpected node kind" );
+		assertEquals( node.getName(), "apartments" );
+		assertEquals( node.as( PropertyNode.class ).getValue(), new Apartment( new Person( "Bob" ) ) );
```

Comment:
I think I would have created the apartment up at the top and added it to the building and used it for the assertion here. But that's a matter of taste manybe.